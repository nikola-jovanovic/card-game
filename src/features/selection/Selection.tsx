import React from 'react'
import Button from '../../shared/components/Button'
import Centered from '../../shared/components/Centered'
import Title from '../../shared/components/Title'

type Props = {
  onSelection: (count: number) => void
}

const playersCount = [2, 3, 4] as const

const Selection = ({ onSelection }: Props): JSX.Element => {

  return (
    <Centered>
      <div>
        <Title>Select number of players</Title>

        {playersCount.map((count) => (
          <Button key={count} onClick={() => onSelection(count)}>
            {count} Players
          </Button>
        ))}
      </div>
    </Centered>
  )
}

export default Selection
