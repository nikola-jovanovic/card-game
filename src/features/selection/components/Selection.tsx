import React from 'react'
import Button from '../../../shared/components/Button'
import CenteredContent from '../../../shared/components/CenteredContent'
import Title from '../../../shared/components/Title'

type Props = {
  onSelection: (count: number) => void
}

const playersCount = [2, 3, 4] as const

const Selection = ({ onSelection }: Props): JSX.Element => {

  return (
    <CenteredContent>
      <div>
        <Title>Select number of players</Title>

        {playersCount.map((count) => (
          <Button key={count} onClick={() => onSelection(count)}>
            {count} Players
          </Button>
        ))}
      </div>
    </CenteredContent>
  )
}

export default Selection
