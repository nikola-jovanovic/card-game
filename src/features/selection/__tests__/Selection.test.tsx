import { fireEvent,render, screen } from '@testing-library/react'
import React from 'react'

import Selection from '../Selection'

describe('Selection', () => {
  it('render', () => {
    const onSelection = jest.fn()
    const { container } = render(<Selection onSelection={onSelection} />)

    const button = screen.getByRole('button', { name: /4 players/i })

    expect(screen.getByRole('heading', { name: /select number of players/i, level: 1 })).toBeInTheDocument()
    expect(screen.getByRole('button', { name: /2 players/i })).toBeInTheDocument()
    expect(screen.getByRole('button', { name: /3 players/i })).toBeInTheDocument()
    expect(button).toBeInTheDocument()
    expect(container).toMatchSnapshot()

    fireEvent.click(button)

    expect(onSelection).toHaveBeenCalledTimes(1)
    expect(onSelection).toHaveBeenCalledWith(4)
  })
})