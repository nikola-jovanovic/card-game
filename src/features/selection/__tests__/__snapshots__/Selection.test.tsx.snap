// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selection render 1`] = `
.c2 {
  background-color: #e5f4e7;
  border-radius: 8px;
  cursor: pointer;
  display: block;
  font-size: 16px;
  margin-bottom: 5px;
  padding: 10px;
  width: 100%;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 10px;
}

.c1 {
  color: #fff;
  margin: 0;
  margin-bottom: 10px;
}

<div>
  <div
    class="c0"
  >
    <div>
      <h1
        class="c1"
      >
        Select number of players
      </h1>
      <button
        class="c2"
      >
        2
         Players
      </button>
      <button
        class="c2"
      >
        3
         Players
      </button>
      <button
        class="c2"
      >
        4
         Players
      </button>
    </div>
  </div>
</div>
`;
